File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/DatabaseHelperInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import com.tinlone.startupexamplejava.dependencies.database.DatabaseHelper;
import com.tinlone.startupexamplejava.dependencies.database.IDatabaseHelper;

import java.util.Collections;
import java.util.List;

public class DatabaseHelperInitializer implements Initializer<IDatabaseHelper> {
    @NonNull
    @Override
    public IDatabaseHelper create(@NonNull Context context) {
        Log.i("loglog", "create: DatabaseHelperInitializer");
        DatabaseHelper.init();
        return DatabaseHelper.getInstance();
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        return Collections.singletonList(LoggerInitializer.class);
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/DatabaseHelperInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import android.util.Log
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.database.DatabaseHelper
import com.tinlone.startupexamplekotlin.dependencies.database.IDatabaseHelper

class DatabaseHelperInitializer : Initializer<IDatabaseHelper?> {
    override fun create(context: Context): IDatabaseHelper {
        Log.i("loglog", "create: DatabaseHelperInitializer")
        DatabaseHelper.init()
        return DatabaseHelper.getInstance()
    }

    override fun dependencies(): List<Class<out Initializer<*>?>> {
        return listOf(LoggerInitializer::class.java)
    }
}